### PR TITLE
Creates iMessage extension for the main app

### PR DESCRIPTION
**Pull request by lauyoo46**
Screenshot 2021-01-28 at 14 49 39
Screenshot 2021-01-28 at 14 49 55
Screenshot 2021-01-28 at 14 52 02

### CODE REVIEW

File: CollectionViewDemo/IconDetailViewController.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    
+    @IBOutlet var nameLabel: UILabel! {
+        didSet {
+            nameLabel.text = icon?.name
+        }
+    }
+    @IBOutlet var descriptionLabel: UILabel! {
+        didSet {
+            descriptionLabel.text = icon?.description
+            descriptionLabel.numberOfLines = 0
```
Comment (czegreanu) [inline CollectionViewDemo/IconDetailViewController.swift]: this should be in storyboard

File: IconDataKit/Icon.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                self.price = Double(value) ?? 0.0
+            }
+        }
+    }
+}
+
+public extension Icon {
+    
+    init?(message: MSMessage?) {
+        guard let messageURL = message?.url else {
```
Comment (czegreanu) [inline IconDataKit/Icon.swift]: concatenate this 2 guards in only one

File: IconDataKit/Icon.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    var queryItems: [URLQueryItem] {
+        var items = [URLQueryItem]()
+        items.append(URLQueryItem(name: QueryItemKey.name.rawValue, value: name))
+        items.append(URLQueryItem(name: QueryItemKey.imageName.rawValue, value: imageName))
+        items.append(URLQueryItem(name: QueryItemKey.description.rawValue, value: description))
+        items.append(URLQueryItem(name: QueryItemKey.price.rawValue, value: String (price)))
+        return items
+    }
+    
+
```
Comment (czegreanu) [inline IconDataKit/Icon.swift]: :(

File: IconStore/IconTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import UIKit
+
+class IconTableViewCell: UITableViewCell {
+    
+    @IBOutlet var iconImageView: UIImageView!
+    @IBOutlet var nameLabel: UILabel!
+    @IBOutlet var descriptionLabel: UILabel! {
+        didSet {
+            descriptionLabel.numberOfLines = 0
```
Comment (czegreanu) [inline IconStore/IconTableViewCell.swift]: same here, move to storyboard

**Review by czegreanu — CHANGES_REQUESTED** [review]

File: CollectionViewDemo/IconDetailViewController.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import IconDataKit
+
+class IconDetailViewController: UIViewController {
+    
+    @IBOutlet var nameLabel: UILabel! {
+        didSet {
+            nameLabel.text = icon?.name
+        }
+    }
+    @IBOutlet var descriptionLabel: UILabel! {
```
Comment (czegreanu) [inline CollectionViewDemo/IconDetailViewController.swift:18]: please leave a space between the fields if they have didSet. it is easier to read

**Review by a-nistor — APPROVED** [review]

**Review by czegreanu — APPROVED** [review]

**Pull request merged**